OpenSim/Framework/Communications/Cache/CachedUserInfo.cs

```diff
@@ -75,6 +75,15 @@ public class CachedUserInfo
         public bool HasReceivedInventory { get { return true; } }
 
         /// <summary>
+        /// This is a list of all of the inventory items that are of type "CallingCard" in the
+        ///   "Calling Card"/Friends/All folder to work around a bug with the viewer creating
+        ///   multiple copies of the same calling card
+        /// </summary>
+        private List<string> _namesOfCallingCardsInCallingCardsFriendsAllFolder = null;
+        private object _namesOfCallingCardsInCallingCardsFriendsAllFolderLock = new object();
```

Comment:
Excuse my ignorance, but couldn't you eliminate `_namesOfCallingCardsInCallingCardsFriendsAllFolderLock` in favor of locking on `_namesOfCallingCardsInCallingCardsFriendsAllFolder`?

IE: the below lock statements would become `lock (_namesOfCallingCardsInCallingCardsFriendsAllFolder)`